internal/plugin/grpc_client_plugin.go

```diff
@@ -221,12 +221,30 @@ func (gc *GrpcClient) sendDataPlaneStatusUpdate(
 		Resource: resource,
 	}
 
-	slog.DebugContext(ctx, "Sending data plane status update request", "request", request)
-	if gc.commandServiceClient == nil {
-		return fmt.Errorf("command service client is not initialized")
+	backOffCtx, backoffCancel := context.WithTimeout(ctx, gc.config.Client.Timeout)
+	defer backoffCancel()
+
+	sendDataPlaneStatus := func() (*v1.UpdateDataPlaneStatusResponse, error) {
+		slog.DebugContext(ctx, "Sending data plane status update request", "request", request)
+		if gc.commandServiceClient == nil {
+			return nil, fmt.Errorf("command service client is not initialized")
```

Comment:
nit: https://github.com/uber-go/guide/blob/master/style.md#error-types
```suggestion
			return nil, errors.New("command service client is not initialized")
```
Technically we should also be exporting the error per the guide.

Reply:
We probably need a series of tasks to adhere to the style guide

Reply:
I made the change from fmt.Errorf to errors.New but didn't externalize the errors